docs/source/tutorial/train_loop.rst

```diff
@@ -0,0 +1,328 @@
+
+        # Check the validation accuracy of prediction after every epoch
+        if train_iter.is_new_epoch:  # If this iteration is the final iteration of the current epoch
+
+            # Display the training loss
+            print('epoch:{:02d} train_loss:{:.04f} '.format(
+                train_iter.epoch, float(to_cpu(loss.data))), end='')
+
+            test_losses = []
+            test_accuracies = []
+            while True:
+                test_batch = test_iter.next()
+                image_test, target_test = concat_examples(test_batch, gpu_id)
+
+                # Forward the test data
+                prediction_test = model(image_test)
+
+                # Calculate the loss
+                loss_test = F.softmax_cross_entropy(prediction_test, target_test)
+                test_losses.append(to_cpu(loss_test.data))
+
+                # Calculate the accuracy
+                accuracy = F.accuracy(prediction_test, target_test)
+                accuracy.to_cpu()
+                test_accuracies.append(accuracy.data)
+
+                if test_iter.is_new_epoch:
+                    test_iter.epoch = 0
+                    test_iter.current_position = 0
+                    test_iter.is_new_epoch = False
+                    test_iter._pushed_position = None
+                    break
+
+            print('val_loss:{:.04f} val_accuracy:{:.04f}'.format(
+                np.mean(test_losses), np.mean(test_accuracies)))
+
+Output
+......
+
+::
+
+    epoch:01 train_loss:0.8072 val_loss:0.7592 val_accuracy:0.8289
+    epoch:02 train_loss:0.5021 val_loss:0.4467 val_accuracy:0.8841
+    epoch:03 train_loss:0.3539 val_loss:0.3673 val_accuracy:0.9007
+    epoch:04 train_loss:0.2524 val_loss:0.3307 val_accuracy:0.9067
+    epoch:05 train_loss:0.4232 val_loss:0.3076 val_accuracy:0.9136
+    epoch:06 train_loss:0.3033 val_loss:0.2910 val_accuracy:0.9167
+    epoch:07 train_loss:0.2004 val_loss:0.2773 val_accuracy:0.9222
+    epoch:08 train_loss:0.2885 val_loss:0.2679 val_accuracy:0.9239
+    epoch:09 train_loss:0.2818 val_loss:0.2579 val_accuracy:0.9266
+    epoch:10 train_loss:0.2403 val_loss:0.2484 val_accuracy:0.9307
+
+6. Save the trained model
+'''''''''''''''''''''''''
+
+Chainer provides two types of :mod:`~chainer.serializers` that can be used to save and restore model state. One supports the HDF5 format and the other supports the NumPy NPZ format. For this example, we are going to use the NPZ
+format to save our model since it is easy to use with NumPy and doesn't need to install any additional dependencies or libraries.
+
+.. testcode::
+
+    serializers.save_npz('my_mnist.model', model)
+
+7. Perform classification by the saved model
+''''''''''''''''''''''''''''''''''''''''''''
+
+Let's use the saved model to classify a new image. In order to load the trained model parameters, we need to perform the
+following two steps:
+
+1. Instantiate the same network as what you trained and obtained the saved weights from.
```

Comment:
I meant the saved weights are obtained from the network. But I guess it's confusing so removed.